arch: nios2: enable System ID soft IP driver

ramakrishnapallala opened the pull request:
The system ID core is a simple read-only device that provides Qsys systems with a unique identifer.

Nios-II processor systems use the system ID core to verify that an executable program was compiled targeting the actual hardware image configured in the target FPGA.

andrewboie commented on samples/boards/altera_max10/sysid/src/main.c:
```diff
@@ -0,0 +1,22 @@
+	s32_t sysid;
+
+	sysid = alt_avalon_sysid_test();
+	if (!sysid)
```
zephyr coding style requires the use of braces at all times

ramakrishnapallala replied:
oops!! sorry i missed it.

dbkinder commented on samples/boards/altera_max10/sysid/README.rst:
```diff
@@ -0,0 +1,34 @@
+=============
+
+nios2 core output
+---------------
```
underline needs to be at least as long as the title (add some more ----s)

ramakrishnapallala replied:
Ok.

dbkinder commented on samples/boards/altera_max10/sysid/src/main.c:
```diff
@@ -0,0 +1,23 @@
+
+#include <zephyr.h>
+#include <misc/printk.h>
+#include "altera_avalon_sysid.h"
```
no such file?

ramakrishnapallala replied:
this sample is only applicable to altera_max10 board. somehow it is being built for qemu_x86 by shippable. I only added altera_max10 under platform_whitelist in sample.yaml so it should not build for other targets.

ramakrishnapallala replied:
@dbkinder now it's fine, shippable is passed.

dbkinder reviewed (changes requested):
+1 for the docs, but there's a missing include file altera_avalon_sysid.h

Commit:
arch: nios2: enable System ID soft IP driver

The system ID core is a simple read-only device that
provides Qsys systems with a unique identifer.

Nios-II processor systems use the system ID core to
verify that an executable program was compiled targeting
the actual hardware image configured in the target FPGA.

Signed-off-by: Ramakrishna Pallala <ramakrishna.pallala@intel.com>

Commit:
tests: boards: add SysID test app for altera_max10 board

Add test application to demonstrate the usage of Nios-II
System ID soft IP.

Signed-off-by: Ramakrishna Pallala <ramakrishna.pallala@intel.com>

andrewboie commented on samples/boards/altera_max10/sysid/src/main.c:
```diff
@@ -0,0 +1,23 @@
+
+	sysid = alt_avalon_sysid_test();
+	if (!sysid) {
+		printk("[SysID] hardware and software appear to be in sync\n");
```
If this is the expected output, can we just turn this into a test case and not a sample, where success/failure is reported via the typical test case way?
We could then remove the build_only tag.

ramakrishnapallala replied:
do you want me to move this to ztest framework?

andrewboie replied:
what I want is for this to be tested at runtime, and report pass/fail like other test cases, and get rid of build_only
This should be a test, not a sample

nashif replied:
put it in tests/boards/....

ramakrishnapallala commented:
@andrewboie , updated the PR please look into it.
